File: SiderShopping/app/src/main/java/com/mmart/sidershopping/view/homepage/HomeViewModel.java
package com.mmart.sidershopping.view.homepage;

import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import com.mmart.sidershopping.model.entity.Banner;
import com.mmart.sidershopping.model.entity.Category;
import com.mmart.sidershopping.model.entity.Product;

import java.util.List;

public class HomeViewModel extends ViewModel {


    public MutableLiveData<List<Banner>> getBanners() {
        HomeRepository homeRepository = new HomeRepository();
        return homeRepository.getBanner();
    }

    public MutableLiveData<List<Category>> getCategories() {
        HomeRepository homeRepository = new HomeRepository();
        return homeRepository.getCategories();
    }

    public MutableLiveData<List<Product>> getBestSellers() {
        HomeRepository homeRepository = new HomeRepository();
        return homeRepository.getBestSellers();
    }
}

File: app/src/main/java/com/caio/challengeandroid/viewmodel/HomeViewModel.kt
package com.caio.lodjinha.viewmodel

import android.util.Log
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.caio.lodjinha.repository.BannerRepository
import com.caio.lodjinha.repository.CategoryRepository
import com.caio.lodjinha.repository.ProductRepository
import com.caio.lodjinha.repository.entity.Banner
import com.caio.lodjinha.repository.entity.Category
import com.caio.lodjinha.repository.entity.Product

class HomeViewModel(
    private val bannerRepository: BannerRepository,
    private val categoryRepository: CategoryRepository,
    private val productRepository: ProductRepository) : ViewModel() {

    private val TAG : String = "HomeViewModel"

    val homeLiveData: MutableLiveData<Home> = MutableLiveData()

    suspend fun loadHomeData(){
        val banner = bannerRepository.getBanner()
        Log.d(TAG, "banner->"+banner.size.toString())

        val category = categoryRepository.getCategory()
        Log.d(TAG, "category->"+category.size.toString())

        val product = productRepository.getProductsMoreSallers()
        Log.d(TAG, "product->"+product.size.toString())

        homeLiveData.postValue(Home(banner,category,product))
    }


    data class Home(val banners: List<Banner>, val categories: List<Category>, val products: List<Product>)
}